### PR TITLE
packet: Ensure app_layer_events is not reused as packet is recycled

### PR DESCRIPTION
**Pull request by jlucovsky**
The app_layer_event packet is allocated once and then the memory is re-used. There are a few places where behavior is conditional on whether it's in use (non-NULL). As packets are recycled, the pointer is never reset to NULL. As a result, incorrect behavior can occur if the pointer value is checked.

This PR changes the packet recycle logic to first free the memory (if used) and then clear the pointer to prevent logic issues.

Link to ticket: https://redmine.openinfosecfoundation.org/issues/7106

Describe changes:
- Removed unused AF-PACKET cpp value (flagged by compilation warning)
- Free and clear the app_layer_event memory
-

### Provide values to any of the below to override the defaults.

- To use an LibHTP, Suricata-Verify or Suricata-Update pull request,
  link to the pull request in the respective `_BRANCH` variable.
- Leave unused overrides blank or remove.

SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

File: src/packet.c [hunk 1]
```diff
@@ -128,7 +128,7 @@ void PacketReinit(Packet *p)
     p->tunnel_rtv_cnt = 0;
     p->tunnel_tpr_cnt = 0;
     p->events.cnt = 0;
```
Comment (inashivb) [inline src/packet.c:130]: Do we need to reset this?

Comment (inashivb) [reply src/packet.c:130]: ah these are diff events. nvm.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 642 | 621 | 96.73% | 

Pipeline 21124

**Review by victorjulien — CHANGES_REQUESTED** [review]
would like to see this fixed elsewhere, like in the mask function that checks the pointer

File: src/source-af-packet.c [hunk 2]
```diff
@@ -257,7 +257,6 @@ static int AFPBypassCallback(Packet *p);
 static int AFPXDPBypassCallback(Packet *p);
 #endif
 
-#define MAX_MAPS 32
```
Comment (catenacyber) [inline src/source-af-packet.c:260]: Nice :-)

**Comment by catenacyber** [comment]
@jlucovsky I am taking on this one for https://github.com/OISF/suricata/pull/11303 with other connex fixes

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11328

**Pull request closed without merging**